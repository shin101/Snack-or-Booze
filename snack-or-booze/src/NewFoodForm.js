import React, {useState} from "react";

function NewFoodForm({ addFood }) {
  const [formData, setFormData] = useState({});
  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData(data => ({
      ...data,
      [name] : value
    }))
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    addFood({...formData});
  }

  // let food = items.find(food => food.id === id);
  // if (!food) return <Redirect to={cantFind} />;

  return (
    <form onSubmit={handleSubmit}>
      <p>Choose food type</p>
      <div id="dropdown">
      <select 
      name="type"
      onChange={handleChange}>
        <option value="snacks">Snacks</option>
        <option value="drinks">Drinks</option>
      </select>
      </div>

      <div id="input">
      <input
      name="name"
      placeholder= "food name here"
      onChange= {handleChange} /></div>
      <button>Add food</button>
    </form>
  );
}

export default NewFoodForm;
