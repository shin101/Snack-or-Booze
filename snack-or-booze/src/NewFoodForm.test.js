import { render, fireEvent, queryByText } from "@testing-library/react";
import React from "react";
import NewFoodForm from "./NewFoodForm";

// smoke test
it("renders without crashing", function(){
  render(<NewFoodForm />);
});

// snapshot test
it("matches snapshot", function (){
  const { asFragment } = render(<NewFoodForm />);
  expect(asFragment()).toMatchSnapshot();
})

it("submits new task correctly", function (){
  const mockAddTask = jest.fn();
  const { queryByText } = render(<NewFoodForm addFood={mockAddTask}/>);
  // eslint-disable-next-line
  const btn = queryByText("Add food");
  fireEvent.click(btn);
  expect(mockAddTask).toHaveBeenCalled();
})