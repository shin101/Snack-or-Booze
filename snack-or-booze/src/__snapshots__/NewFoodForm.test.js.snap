// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <div
      id="dropdown"
    >
      <select
        name="type"
      >
        <option
          value="snacks"
        >
          Snacks
        </option>
        <option
          value="drinks"
        >
          Drinks
        </option>
      </select>
    </div>
    <div
      id="foodname"
    >
      <input
        name="name"
        placeholder="food name here"
      />
    </div>
    <div
      id="description"
    >
      <input
        name="description"
        placeholder="description"
      />
    </div>
    <div
      id="recipe"
    >
      <input
        name="recipe"
        placeholder="recipe here"
      />
    </div>
    <div
      id="serve"
    >
      <input
        name="serve"
        placeholder="serve info"
      />
    </div>
    <button>
      Add food
    </button>
  </form>
</DocumentFragment>
`;
